tus/src/main/java/ch/cyberduck/core/tus/TusUploadFeature.java

```diff
@@ -0,0 +1,243 @@
+                        .segment(true)
+                        .withOffset(offset)
+                        .withLength(length);
+                status.setHeader(overall.getHeader());
+                status.setChecksum(writer.checksum(file, status).compute(local.getInputStream(), status));
+                final Map<String, String> parameters = new HashMap<>();
+                parameters.put(UPLOAD_URL, uploadUrl);
+                status.withParameters(parameters);
+                final Void response = TusUploadFeature.this.upload(
+                        file, local, throttle, listener, status, overall, status, callback);
+                if(log.isInfoEnabled()) {
+                    log.info(String.format("Received response %s", response));
+                }
+                return null;
+            }
+        }, overall).call());
+    }
+
+    @Override
+    public Write.Append append(final Path file, final TransferStatus status) throws BackgroundException {
+        // Determine the offset at which the upload should be continued
+        final String property = toProperty(host, file);
+        final String uploadUrl = preferences.getProperty(property);
+        if(StringUtils.isBlank(uploadUrl)) {
+            if(log.isDebugEnabled()) {
+                log.debug(String.format("No previous upload URL for %s", file));
+            }
+            return Write.override;
+        }
+        final HttpHead request = new HttpHead(uploadUrl);
+        request.setHeader(TUS_HEADER_RESUMABLE, TUS_VERSION);
+        try {
+            final Long offset = client.execute(request, new ResponseHandler<Long>() {
+                @Override
+                public Long handleResponse(final HttpResponse response) throws HttpResponseException {
+                    switch(response.getStatusLine().getStatusCode()) {
+                        case HttpStatus.SC_OK:
+                            if(response.containsHeader(TUS_HEADER_UPLOAD_OFFSET)) {
+                                final Header header = response.getFirstHeader(TUS_HEADER_UPLOAD_OFFSET);
+                                if(log.isDebugEnabled()) {
+                                    log.debug(String.format("Return offset header %s", header));
+                                }
+                                return Long.valueOf(header.getValue());
+                            }
+                    }
+                    // If the resource is not found, the Server SHOULD return either the 404 Not Found,
+                    // 410 Gone or 403 Forbidden status without the Upload-Offset header.
+                    throw new HttpResponseException(response.getStatusLine().getStatusCode(), response.getStatusLine().getReasonPhrase());
+                }
+            });
+            return new Write.Append(true).withStatus(status).withSize(offset);
+        }
+        catch(HttpResponseException e) {
+            preferences.deleteProperty(property);
```

Comment:
Check for status code 404, 410 or 403.

Reply:
Would rather not depend on. The tus implementation of ownCloud returns arbitrary status codes such as `502`.

Reply:
The spec says _should_ return 404, 410 or 403 but not _must_. Makes sense then.